code/game/atoms.dm

```diff
@@ -1842,11 +1842,13 @@
 		if(!length(forced_gravity))
 			SEND_SIGNAL(gravity_turf, COMSIG_TURF_HAS_GRAVITY, src, forced_gravity)
 
-		var/max_grav = 0
-		for(var/i in forced_gravity)//our gravity is the strongest return forced gravity we get
-			max_grav = max(max_grav, i)
-		//cut so we can reuse the list, this is ok since forced gravity movers are exceedingly rare compared to all other movement
-		return max_grav
+		var/positive_grav = 0
+		var/negative_grav = 0
+		for(var/gravity_influence in forced_gravity)//our gravity is the strongest return forced gravity we get
+			positive_grav = max(positive_grav, gravity_influence)
```

Comment:
```suggestion
		var/positive_grav = 0
		var/negative_grav = 0
		//our gravity is sum of the most massive positive and negative numbers returned by the signal
		//so that adding two forced_gravity elements with an effect size of 1 each doesnt add to 2 gravity
		//but negative force gravity effects can cancel out positive ones
		for(var/gravity_influence in forced_gravity)
			positive_grav = max(positive_grav, gravity_influence)
```